src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IDynamicMemberReferenceExpression.cs

```diff
@@ -436,5 +436,236 @@ Type Arguments(0)
 
             VerifyOperationTreeAndDiagnosticsForTest<MemberAccessExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
         }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void DynamicMemberReference_ControlFlowInSimpleMemberAccess()
+        {
+            string source = @"
+using System;
+
+namespace ConsoleApp1
+{
+    class C1
+    {
+        static void M1(dynamic d, dynamic p)
+        /*<bind>*/{
+            p = d?.Prop1;
```

Comment:
>d?.Prop1 [](start = 16, length = 8)

It doesn't look like any of the tests are actually testing interesting scenarios with IDynamicMemberReferenceOperation. All they test conditional access and it is not inside IDynamicMemberReferenceOperation, but rather the IDynamicMemberReferenceOperation is inside conditional access. Please replace them with tests that are targeting control flow inside IDynamicMemberReferenceOperation. For example,  ```(d1 ?? d2).Prop1```. We also should have a simple test that doesn't have any control flow.

Reply:
I'd say we need only two tests:
- no control flow
- control flow in the receiver

---
In reply to: [186833448](https://github.com/dotnet/roslyn/pull/26690#discussion_r186833448) [](ancestors = 186833448)

Reply:
Sure, will update.

---
In reply to: [186834523](https://github.com/dotnet/roslyn/pull/26690#discussion_r186834523) [](ancestors = 186834523,186833448)

Reply:
We also need tests covering non-empty TypeArguments and Containing type, could be folded into no control flow test
Also, null receiver, if possible

---
In reply to: [186834783](https://github.com/dotnet/roslyn/pull/26690#discussion_r186834783) [](ancestors = 186834783,186834523,186833448)

Reply:
Updated to have only two tests. Null receiver is covered in test `DynamicInvocation_ControlFlowNullReceiver`

---
In reply to: [186837885](https://github.com/dotnet/roslyn/pull/26690#discussion_r186837885) [](ancestors = 186837885,186834783,186834523,186833448)